library/Fission/CLI/Command/Watch.hs

```diff
@@ -64,18 +87,36 @@ watcher :: MonadRIO          cfg m
         => HasProcessContext cfg
         => Has IPFS.BinPath  cfg
         => Has IPFS.Timeout  cfg
-        => String
+        => CommandOptions
         -> m ()
-watcher dir = handleWith_ CLI.Error.put' do
+watcher options = handleWith_ CLI.Error.put' do
+  let dir = optLocation options
+  let dnsOnly = optDnsOnly options
+
   cfg <- ask
   curr <- getCurrentDirectory
+
   let absDir = if isAbsolute dir then dir else curr </> dir
   UTF8.putText $ "👀 Watching " <> Text.pack dir <> " for changes...\n"
```

Comment:
As a general pattern, we've been keeping assignment together (which I find to be useful in any langauge)

# Before

```haskell
watcher options = handleWith_ CLI.Error.put' do
  let dir = optLocation options
  let dnsOnly = optDnsOnly options

  cfg <- ask
  curr <- getCurrentDirectory

  let absDir = if isAbsolute dir then dir else curr </> dir
  UTF8.putText $ "👀 Watching " <> Text.pack dir <> " for changes...\n"
```

# After

```haskell
watcher options = handleWith_ CLI.Error.put' do
  -- Effects --
  cfg <- ask
  curr <- getCurrentDirectory

  -- Local Assignment --
  let 
    dir = optLocation options
    dnsOnly = optDnsOnly options
    absDir = if isAbsolute dir then dir else curr </> dir

  -- Do Stuff --
  UTF8.putText $ "👀 Watching " <> Text.pack dir <> " for changes...\n"
```